22-discovery.ks

```diff
@@ -48,7 +48,10 @@ systemctl disable ipmi.service
 echo " * open foreman-proxy port via firewalld"
 firewall-offline-cmd --zone=public --add-port=8443/tcp --add-port=8448/tcp
 
-echo " * setting up foreman proxy"
+echo " * setting up foreman proxy service"
+sed -i 's/After=.*/After=basic.target network-online.target nm-prepare.service/' /usr/lib/systemd/system/foreman-proxy.service
+sed -i 's/Wants=.*/Wants=basic.target network-online.target nm-prepare.service/' /usr/lib/systemd/system/foreman-proxy.service
+sed -i '/\[Unit\]/a ConditionPathExists=/etc/NetworkManager/system-connections/primary' /usr/lib/systemd/system/foreman-proxy.service
```

Comment:
Makes sure this starts really when Network Online target reaches "started" state. Also skips this service when doing PXE-less boot (primary connection is not configured in that case).